spring-cloud-stream/src/main/java/org/springframework/cloud/stream/provisioning/ProducerDestination.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.provisioning;
+
+import org.springframework.cloud.stream.binder.ProducerProperties;
+
+/**
+ * Represents a ProducerDestination that provides the information about the destination
+ * that is physically provisioned through
+ * {@link ProvisioningProvider#provisionProducerDestination(String, ProducerProperties)}
+ *
+ * @author Soby Chacko
+ *
+ * @since 1.2
+ */
+public interface ProducerDestination {
+
+	/**
+	 * Provides the destination name.
+	 *
+	 * @return destination name
+	 */
+	String getName();
+
+	/**
+	 * Provides the destination name for a given partition.
+	 *
+	 * If the producer provision the destination with partitions, on certain middleware brokers
+	 * there may exist multiple destinations distinguishable by the partition. For example,
+	 * if the destination name is <b>xyz</b> and is is provisioned with <b>4</b> partitions, there may be
+	 * 4 different destinations on the broker such as - <b>xyz-0, xyz-1, xyz-2 and xyz-3</b>.
+	 * However, please keep in mind that, this behavior is completely dependent on the broker
```

Comment:
M-m-m. Not sure in this `please`.
The JavaDoc is fully official text and should just describe facts about the API.
In this case to say that `A behavior is completely...` is pretty enough